githubkit/core.py

```diff
@@ -323,33 +324,64 @@ def _check(
             error_models = error_models or {}
             status_code = str(response.status_code)
 
-            if response.status_code == 403 and "retry-after" in response.headers:
-                raise RateLimitExceededError(
-                    timedelta(seconds=int(response.headers["retry-after"])),
-                    original_message=response.content,
-                )
-
+            error_model = error_models.get(
+                status_code,
+                error_models.get(
+                    f"{status_code[:-2]}XX", error_models.get("default", Any)
+                ),
+            )
+            resp = Response(response, error_model)
+
+            # handle rate limit exceeded
```

Comment:
This method is getting quite big maybe extract the rate limit checks to `_check_rate_limit_exceeded` (or something with a different name)